metrics/core/src/main/java/org/hyperledger/besu/metrics/prometheus/PrometheusGauge.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.metrics.prometheus;
+
+import org.hyperledger.besu.plugin.services.metrics.LabelledGauge;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.DoubleSupplier;
+
+import io.prometheus.client.Collector;
+
+public class PrometheusGauge implements LabelledGauge {
+  private final String metricName;
+  private final String help;
+  private final List<String> labelNames;
+  private final Consumer<Collector> collectorConsumer;
+  private final List<String> labelValuesCreated = new ArrayList<>();
```

Comment:
This won't be thread safe, probably want to use a `ConcurrentHashSet`.  As with the open telemetry version need to try adding and throw if it already exists rather than doing the contains and add steps separately.